compiler-k2/src/test/kotlin/com/squareup/anvil/compiler/k2/fir/constructor/inject/FirInjectConstructorFactoryGenerationExtensionTest.kt

```diff
@@ -0,0 +1,61 @@
+package com.squareup.anvil.compiler.k2.fir.constructor.inject
+
+import com.squareup.anvil.compiler.testing.CompilationMode
+import com.squareup.anvil.compiler.testing.CompilationModeTest
+import dagger.internal.Provider
+import io.kotest.matchers.nulls.shouldBeNull
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.TestFactory
+
+class FirInjectConstructorFactoryGenerationExtensionTest : CompilationModeTest(
+  CompilationMode.K2(useKapt = false),
+) {
+  @TestFactory
+  fun `factory class is generated for @Inject annotation`() = params.asTests {
+    compile2(
+      """
+      package foo
+      class InjectClass @javax.inject.Inject constructor()
+      """.trimIndent(),
+    ) {
+      classGraph.allClassesAsMap["foo.InjectClass_Factory"].shouldNotBeNull()
+    }
+  }
+
+  @TestFactory
+  fun `factory class is not generated if class is not annotated`() = params.asTests {
+    compile2(
+      """
+      package foo
+      class InjectClass()
+      """.trimIndent(),
+    ) {
+      classGraph.allClassesAsMap["foo.InjectClass_Factory"].shouldBeNull()
+    }
+  }
+
+  @TestFactory
+  fun `factory class creates source class with values from provider`() = params.asTests {
+    compile2(
+      """
+      package foo
+      class InjectClass @javax.inject.Inject constructor(
+        val param0: String,
+        val param1: Int,
+      )
+      """.trimIndent(),
+    ) {
+      val expectedString = "ExpectedString"
+      val expectedInt = 77
+      val injectFactoryClass = classGraph.allClassesAsMap["foo.InjectClass_Factory"]!!.loadClass()
+      val factoryInstance = injectFactoryClass.constructors.first()
+        .newInstance(Provider { expectedString }, Provider { expectedInt })
```

Comment:
Should we be calling the companion object's `create` method instead here to verify everything was generated, either here or as a separate test? I think right now all of the tests would continue to pass even if we were no longer generating the `create` method

Reply:
Ah yes good call I'll update the test to use the static create method